src/test/scala/strawman/collection/test/TraverseTest.scala

```diff
@@ -1,77 +1,139 @@
-//package strawman
-//
-//package collection.test
-//
-//import org.junit.Test
-//import strawman.collection.Iterable
-//import strawman.collection.mutable.Builder
-//import strawman.collection._
-//
-//import scala.{Any, Either, Int, Left, None, Option, Right, Some, Unit}
-//import scala.Predef.ArrowAssoc
-//import scala.math.Ordering
-//import java.lang.String
-//
-//class TraverseTest {
-//
-//  // You can either overload methods for PolyBuildable and OrderedPolyBuildable (if you want to support ordered collection types)
-//  def optionSequence1[C[X] <: Iterable[X] with PolyBuildable[X, C], A](xs: C[Option[A]]): Option[C[A]] =
-//    xs.foldLeft[Option[Builder[A, C[A]]]](Some(xs.newBuilder)) {
-//      case (Some(builder), Some(a)) => Some(builder += a)
-//      case _ => None
-//    }.map(_.result)
-//
-//  def optionSequence1[CC[_], A](xs: OrderedPolyBuildable[Option[A], CC] with Iterable[Option[A]])(implicit ev: Ordering[A]): Option[CC[A]] =
-//    xs.foldLeft[Option[Builder[A, CC[A]]]](Some(xs.newOrderedBuilder)) {
-//      case (Some(builder), Some(a)) => Some(builder += a)
-//      case _ => None
-//    }.map(_.result)
-//
-//  // ...or use BuildFrom to abstract over both and also allow building arbitrary collection types
-//  def optionSequence[CC[X] <: Iterable[X], A](xs: CC[Option[A]])(implicit bf: BuildFrom[CC[Option[A]], A]): Option[bf.To] =
-//    xs.foldLeft[Option[Builder[A, bf.To]]](Some(bf.newBuilder(xs))) {
-//      case (Some(builder), Some(a)) => Some(builder += a)
-//      case _ => None
-//    }.map(_.result)
-//
-//  def eitherSequence[C[X] <: Iterable[X], A, B](xs: C[Either[A, B]])(implicit bf: BuildFrom[xs.type, B]): Either[A, bf.To] =
-//    xs.foldLeft[Either[A, Builder[B, bf.To]]](Right(bf.newBuilder(xs))) {
-//      case (Right(builder), Right(b)) => Right(builder += b)
-//      case (Left(a)       ,        _) => Left(a)
-//      case (_             ,  Left(a)) => Left(a)
-//    }.right.map(_.result)
-//
+package strawman
+
+package collection.test
+
+import org.junit.Test
+import strawman.collection.Iterable
+import strawman.collection.mutable.Builder
+import strawman.collection._
+
+import scala.{Any, Either, Int, Left, None, Option, Right, Some, Unit}
+import scala.Predef.{ArrowAssoc, implicitly}
+import scala.math.Ordering
+import java.lang.String
+
+/**
+  * Auxiliary type allowing us to define which collection type to build according to the type of an existing collection.
+  * @tparam From Existing collection type that will drives the collection type to build
+  * @tparam A Element type
+  */
+trait BuildFrom[-From, -A] {
+
+  /** Collection type to build */
+  type To
+
+  def newBuilder(): Builder[A, To]
+
+}
```

Comment:
I re-introduced the `BuildFrom` type initially written by @szeiger. This type is used to infer the type of a collection to build (`To`) based on the type of an existing collection (`From`).

I think this use case is very specific to `Traverse` where we want to turn a `F[G[?]]` into a `G[F[?]]`, so I think it makes sense to not include it in the core of the collections (well, it will be in the standard library anyway if we want to keep `Future.sequence`).